src/Microsoft.NetCore.Analyzers/Core/Security/SourceTriggeredTaintedDataAnalyzerBase.cs

```diff
@@ -0,0 +1,133 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+namespace Microsoft.NetCore.Analyzers.Security
+{
+    using System.Collections.Generic;
+    using System.Collections.Immutable;
+    using System.Linq;
+    using Analyzer.Utilities.Extensions;
+    using Analyzer.Utilities.FlowAnalysis.Analysis.TaintedDataAnalysis;
+    using Microsoft.CodeAnalysis;
+    using Microsoft.CodeAnalysis.Diagnostics;
+    using Microsoft.CodeAnalysis.Operations;
+
+    /// <summary>
+    /// Base class to aid in implementing tainted data analyzers.
+    /// </summary>
+    public abstract class SourceTriggeredTaintedDataAnalyzerBase : DiagnosticAnalyzer
```

Comment:
>SourceTriggeredTaintedDataAnalyzerBase [](start = 26, length = 38)

Presuming this is just moving existing code to a new type and so skipping reviewing it - let me know if that is not the case. Thanks!

Reply:
Ah, just noticed you moved to use of TaintedDataConfig here. Seems fine to me.

---
In reply to: [239837489](https://github.com/dotnet/roslyn-analyzers/pull/1910#discussion_r239837489) [](ancestors = 239837489)